### PR TITLE
build: update gradle wrapper to 7.6

### PR DESCRIPTION
**Pull request by yeikel**
I generated this pull request using the wrapper itself

` ./gradlew wrapper --gradle-version 7.6`

> 
> Here are the highlights of this release:
>  - Added support for Java 19.
>  - Introduced `--rerun` flag for individual task rerun.
>  - Improved dependency block for test suites to be strongly typed.
>  - Added a pluggable system for Java toolchains provisioning.
> 
> For more details see https://docs.gradle.org/7.6/release-notes.html

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
+distributionSha256Sum=7ba68c54029790ab444b39d7e293d3236b2632631fb5f2e012bb28b4ff669e4b
```
Comment (yeikel) [inline gradle/wrapper/gradle-wrapper.properties:4]: See #2648

File: rewrite-java-11/build.gradle.kts [hunk 2]
```diff
@@ -57,7 +57,7 @@ testing {
 
         register("compatibilityTest", JvmTestSuite::class) {
             dependencies {
-                implementation(project)
+                implementation(project())
```
Comment (yeikel) [inline rewrite-java-11/build.gradle.kts:60]: See https://github.com/gradle/gradle/issues/22720

File: gradlew.bat [hunk 3]
```diff
@@ -25,7 +25,7 @@
 if "%OS%"=="Windows_NT" setlocal
 
 set DIRNAME=%~dp0
-if "%DIRNAME%" == "" set DIRNAME=.
+if "%DIRNAME%"=="" set DIRNAME=.
 set APP_BASE_NAME=%~n0
 set APP_HOME=%DIRNAME%
```
Comment (yeikel) [inline gradlew.bat:31]: This was generated by the wrapper generation

**Comment by sambsnyd** [comment]
Thanks @yeikel !

**Pull request merged**